My first draft of addressing the queries

drvinceknight opened the pull request:
The biggest change is the restructure: I have moved a bunch of the examples to the Reuse section and added a paragraph to the Architecture section.

marcharper commented on tex/paper.tex:
```diff
@@ -125,15 +125,15 @@ \section*{Keywords}
+2012 tournament \cite{Stewart2012}, following the discovery of zero-determinant
+strategies.  Subsequent research has spawned a number of papers (many of which
+are referenced throughout this paper), but rarely are the results reproducible.
+Amongst well-known tournaments, in only one case is the full original source code
```
I prefer among, amongst sounds pretentious.

drvinceknight replied:
This was specifically requested by one of the reviewers.  I personally don't care but I suggest we go with their preference?

marcharper replied:
I actually googled "among vs amongst" and most sources say they are interchangeable. I guess it's a minor thing.

marcharper commented on tex/paper.tex:
```diff
@@ -158,20 +157,21 @@ \section*{Introduction}
-        strategies available in the case of parametrised strategies
-    \item Extensible: easy to modify to include new strategies and to run new tournaments
+    \item Extensive: 135 strategies are included, with infinitely-many
+        available in the case of parametrised strategies
```
"Parameterized" is more common on this side of the pond but I guess I'm outnumbered here :)

drvinceknight replied:
This might come up at the editorial/proofing stage. It's happened to me before at that stage where all things went one way or the other... I'm again indifferent...

marcharper replied:
Ok we'll see what they want in proofing.

meatballs replied:
either way, it's spelt wrong. "parameterised" (or "parameterized")

meatballs replied:
Seems I'm wrong - my apologies. Never seen that spelling before!

marcharper commented on tex/paper.tex:
```diff
@@ -302,54 +302,17 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
-classes). Full contribution guidelines can be found in the documentation.
-As an example, Figures~\ref{fig:grudger} and~\ref{fig:grudger_test} show the source code for
+classes). All components of the library are automatically tested using a
+combination of unit, property and integration tests, these tests are run as new
```
Oxford comma is better here? Also this seems like a run-on sentence.

marcharper commented on tex/paper.tex:
```diff
@@ -302,54 +302,17 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+classes). All components of the library are automatically tested using a
+combination of unit, property and integration tests, these tests are run as new
+features are added to the library to ensure compatibility (they are also run
+automatically using \url{travis-ci.org}). Thus, when submitting a strategy, a
```
~~Thus~~

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+You can see an overview of the structure of the source code in
+Figure~\ref{fig:overview}. This shows the parallel collection of strategies and
+their tests. Furthermore the underlying engine for the library is a class for
+tournaments which lives in the \texttt{tournament.py} module. This class is
```
not technically a module but that's fine

drvinceknight replied:
Is it not? I thought that's what a module was...

From:

https://www.google.co.uk/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=what%20is%20a%20python%20module

> A module is a Python object with arbitrarily named attributes that you can bind and reference. Simply, a module is a file consisting of Python code. A module can define functions, classes and variables. A module can also include runnable code.

Not defending, just asking what a module is :)

marcharper replied:
I thought a module had to have its own folder and an `__init__` file, but it looks like just a file is fine.

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+tournaments which lives in the \texttt{tournament.py} module. This class is
+responsible for coordinating the play of generated matches (from the
+\texttt{match.py} module). This generation of matches is the responsibility of a
+match generator class (in the \texttt{match\_generator.py} module) which is
```
maybe quotes or italicize `match generator`

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+\texttt{match.py} module). This generation of matches is the responsibility of a
+match generator class (in the \texttt{match\_generator.py} module) which is
+designed in such a way as to be easily modifiable to create new types of
+tournaments. This is described fuller in a tutorial in the documentation which
```
~~fuller~~ further? more fully?

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+\end{figure}
+
+To date the library has had contributions from 25 contributors from a variety
+of backgrounds which are not solely academic. These contributions have been in
```
_mostly_ in terms of strategies

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+strategies were written by a 15 year old secondary school student. Both of
+these students are authors of this paper. As well as these strategy
+contributions, vital architectural improvements to the library itself have also
+been received.
```
~~received~~ contributed?

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+
+\section*{Programming language}
+
+The library is continuously tested for compatibility with Python 2.7, 3.4 and 3.5.
```
and the two most recent python 3 releases

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+    \begin{itemize}[noitemsep,topsep=0pt]
+        \item Numpy 1.9.2
+        \item Matplotlib 1.4.2
+        \item Testfixtures 4.9.1
```
is testfixtures required to run the library or also just for development?
also tqdm is optional? as is matplotlib?

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+
+\section*{Reuse potential}\label{sec:reuse}
+
+The Axelrod library has been designed with sustainable software practice in
```
practices

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+of example Jupyter notebooks available here:
+\url{https://github.com/Axelrod-Python/Axelrod-notebooks}.
+
+The ability to have readily available a large number of strategies makes this
```
This sentence is awkward... maybe something like: "The availability of a large number of strategies...

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+
+The ability to have readily available a large number of strategies makes this
+tool an excellent and obvious example of the benefits of open research which
+should positively impact the game theoretic community.
```
game theory community

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+The ability to have readily available a large number of strategies makes this
+tool an excellent and obvious example of the benefits of open research which
+should positively impact the game theoretic community.
+This has been evidenced by the work described: the library has already shown
```
This is evidently true already as the library has been used to study and create interesting and powerful new strategies.

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+This has been evidenced by the work described: the library has already shown
+its potential for reuse.
+
+Installation of the library is straightforward as it is available via the
```
straightforward via standard python installation repositories...

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+standard Python installation package: `pip'
+(\url{https://pypi.python.org/pypi}). The package name is \texttt{axelrod} and
+can thus be installed by calling: \texttt{pip install axelrod}. This ensures it
+can be used on all major operating systems (Windows, OS X and Linux).
```
Technically availability through pip doesn't require compatibility with all OSes.

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+	\label{fig:number_of_strategies_against_date}
+\end{figure}
+
+Despite this, due to previous research being done in an irreproducible manner
```
~~despite this~~: Nevertheless

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+Despite this, due to previous research being done in an irreproducible manner
+with, for example, no source code and/or vaguely described strategies, not all
+previous tournaments can yet be reproduced. In fact, some of the early
+tournaments might be impossible to reproduce as the source code is forever
```
apparently forever lost

marcharper commented on tex/paper.tex:
```diff
@@ -411,41 +373,195 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+with, for example, no source code and/or vaguely described strategies, not all
+previous tournaments can yet be reproduced. In fact, some of the early
+tournaments might be impossible to reproduce as the source code is forever
+lost. This library aims to prevent this from happening again in the future.
```
prevent what? How about "ensure reproducibility in the future".

marcharper commented on tex/paper.tex:
```diff
@@ -475,12 +591,19 @@ \section*{New strategies, tournaments and implications}\label{sec:new-strategies
 This can be reproduced as shown in Figure~\ref{fig:stewart-code}, which gives
-the plot of Figure~\ref{fig:stewart_tournament}.
+the plot of Figure~\ref{fig:stewart_tournament}. Note that slight differences
+with the results of \cite{Stewart2012} are due to stochastic behaviour.
```
of some strategies.

marcharper commented on tex/paper.tex:
```diff
@@ -512,16 +636,29 @@ \section*{New strategies, tournaments and implications}\label{sec:new-strategies
-detail) here: \url{http://axelrod-tournament.readthedocs.org/}.
+detail) here: \url{http://axelrod-tournament.readthedocs.org/}. Data sets are
+also available showing the plays of every match that takes place. Note that to
+obtain this tournament requires a simple change of the code shown in
```
recreate this tournament.... simply requires changing a single line...

marcharper commented on tex/paper.tex:
```diff
@@ -512,16 +636,29 @@ \section*{New strategies, tournaments and implications}\label{sec:new-strategies
 \end{figure}
 
-The current winning strategy is a novel strategy called: Looker Up. This is a
+The current winning strategy is a novel strategy: Looker Up. This is a
```
is "new to the research literature"

drvinceknight commented:
Just to say I've seen these @marcharper. Really great, should get to them tomorrow. :+1:

drvinceknight commented:
Have gone through and made most changes. One of the changes need to check that https://github.com/Axelrod-Python/Axelrod/pull/622 passes travis.

meatballs commented on tex/paper.tex:
```diff
@@ -125,15 +125,15 @@ \section*{Keywords}
+with significant test suites.
 
 To make matters more complicated, often a newly-created strategy is studied in
 isolation by opponents chosen by the strategy's creators, and often such
```
'with opponents' rather than 'by opponents'

meatballs commented on tex/paper.tex:
```diff
@@ -145,9 +145,8 @@ \section*{Introduction}
 cannot be reliably replicated, and therefore have not met the basic scientific
 criterion of falsifiability.
 
```
Suggest instead:

To complicate matters further, a new strategy is often studied in isolation with opponents chosen by the creator of that strategy. As such, the results cannot be reliably replicated and therefore have not met the basic scientific criterion of falsifiability.

meatballs commented on tex/paper.tex:
```diff
@@ -411,41 +373,196 @@ \section*{Description of the Axelrod Python package}\label{sec:description-of-ax
+        \item {Jason Young}
+        \item {Andy Boot}
+        \item {Anna Barriscale}
+    \end{itemize}
```
Nikoleta

drvinceknight commented:
Have addressed your comments (I think) @meatballs. Things left to do (this list is for my benefit):
- [x] Rerun scraping of the github repo (new strategy from Nikoleta)
- [x] Update full tournament plot (new strategy from Nikoleta)
- [x] Have a final check over the response document

drvinceknight commented:
If https://github.com/Axelrod-Python/Axelrod/pull/628 is merged before we submit, I could update the results again...

meatballs commented on tex/response.tex:
```diff
@@ -0,0 +1,85 @@
+Furthermore, we have made some minor changes to the text. These were all minor
+typographical changes aimed at improving the readability of the manuscript.
+
+\end{document}
```
I've looked through the response and it's all fine with me.

drvinceknight commented:
Have added @314pe to the authors. Just need to update graphs and this is good to go. 👍

drvinceknight commented:
One last thing is to update the release. Waiting on zenodo.

langner commented:
I read through everything and it looks good to me. You guys are amazing!

On Mon, Jun 13, 2016 at 8:55 PM, Vince Knight notifications@github.com
wrote:

> One last thing is to update the release. Waiting on zenodo.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/Axelrod-Python/An-open-reproducible-framework-for-the-study-of-the-iterated-prisoners-dilemma/pull/24#issuecomment-225674715,
> or mute the thread
> https://github.com/notifications/unsubscribe/AAtZEtA6C-7kGeqkHBrOpW4WrzbYZqvYks5qLae3gaJpZM4ItwpH
> .
